netket/graph/grid.py

```diff
@@ -43,6 +43,58 @@ def __hash__(self):
         return hash((self.shifts, self.dims))
 
 
+@dataclass
```

Comment:
Just noticed that myself: Elements that are `dataclass`es should be frozen (that is, immutable) in order to have a well-defined hash.

On the positive side, if `frozen=True` is passed, the `__hash__` method is created automatically.

```suggestion
@dataclass(frozen=True)
```